### PR TITLE
osxfuse deleted

### PR DESCRIPTION
**Pull request by AnastasiaSulyagina**
See [#15603](https://github.com/caskroom/homebrew-cask/issues/15603) for more details

### CODE REVIEW

**Comment by ilovezfs** [comment]
I'm starting to think there should be a kext tap. This PR makes me want to :cry:.

**Comment by MikeMcQuaid** [comment]
@ilovezfs You could create one but not inside the Homebrew org as they aren't supported on 10.10 and above.

**Comment by DomT4** [comment]
We're going to lose the ability to CI test them at all in a few months, and generally, nudging towards signed kernel extensions feels like a good thing to encourage people to do. Agree `tuntap`, `osxfuse`, there's a couple of others I think, need to die off in favour of the Casks.

We'll need to rework our Requirements involved here before these can actually 💀, but I presume Anastasia is on top of that.

**Comment by ilovezfs** [comment]
The fact that you can rebuild your kernel cache with an usigned kext *and keep SIP enabled* seems to be missed here.

**Comment by DomT4** [comment]
That's not something we support, though. Frankly, the previous default where kernel extensions didn't have to be signed by upstream was a risky default & I'm glad Apple changed that. "Don't blindly trust kernel extensions" seems a pretty reasonable policy to me.

**Comment by ilovezfs** [comment]
It strikes me as quiet short-sighted to cut Homebrew off from the entire domain of xnu kernel extension development as if it doesn't exist.

**Comment by DomT4** [comment]
Is already the case for ~89% of Homebrew's userbase, which zero complaints about that to date. We've not exactly had people banging down the door in the last two years saying _"Please change this!!!"_.

If people want to support it in external taps, that's fine, we don't police those, but enabling the use of unsigned kernel extensions where a solid majority of users won't know or consider the risk factor was never a great policy to begin with.

**Comment by MikeMcQuaid** [comment]
> If people want to support it in external taps, that's fine, we don't police those, but enabling the use of unsigned kernel extensions where a solid majority of users won't know or consider the risk factor was never a great policy to begin with.

Completely agree. It'd be good if we could reserve "why is Homebrew stopping doing X" for stuff we're making it impossible for third parties (or maintainers on taps they run outside the Homebrew org) to do rather than the stuff we're not maintaining any more.

**Comment by MikeMcQuaid** [comment]
@Homebrew/maintainers Any final (very strong) objections before I merge this?

**Comment by DomT4** [comment]
We need to fix the Requirements first. That's all from me.

**Comment by MikeMcQuaid** [comment]
Cool, thanks @DomT4. @AnastasiaSulyagina we'll talk more about that later.

**Comment by MikeMcQuaid** [comment]
Requirements fix in https://github.com/Homebrew/brew/pull/487.

File: tap_migrations.json [hunk 1]
```diff
@@ -279,5 +279,6 @@
   "xspringies": "homebrew/x11",
   "xulrunner": "homebrew/boneyard",
   "yarp": "homebrew/x11",
-  "ydict": "homebrew/boneyard"
+  "ydict": "homebrew/boneyard",
+  "osxfuse": "caskroom/cask"
```
Comment (MikeMcQuaid) [inline tap_migrations.json]: Can you make sure this is in alphabetical order? Thanks!

**Comment by MikeMcQuaid** [comment]
This PR needs rebased on `master` (I suspect there's been a formula change).

**Comment by MikeMcQuaid** [comment]
@AnastasiaSulyagina 😍

**Pull request merged**